appointment-service/src/main/java/com/example/AppointmentService/Schedule/IncomingEvent.java

```diff
@@ -48,12 +46,15 @@ public void onEvent(Map<String, Object> headers, String payload) {
                 Object lotSize = msgMap.get("lotSize");
                 Object centerId = msgMap.get("centerId");
 
-                Schedule schedule = scheduleRepository.findByVaccineTypeId((int) vaccineId).orElseThrow(() -> new ResourceNotFoundException("Schedule not found with vaccine id " +vaccineId));
-                List<Appointment> appointmentList = new ArrayList<>();
-                for(int i = 0; i<(int)lotSize;i++){
-                    appointmentList.add(new Appointment(schedule,0,(int) centerId,"none","none","open",null));
-                }
-                appointmentRepository.saveAll(appointmentList);
+                Schedule schedule = scheduleRepository.findFirstByOrderByDateDesc().orElseThrow(() -> new ResourceNotFoundException("Schedule not found"));
+                Date date = schedule.getDate();
+                Calendar calendar = Calendar.getInstance();
+                calendar.setTime(date);
+                calendar.add(Calendar.DATE,1);
+                Date newDate = new Date(calendar.getTimeInMillis());
+
+                Schedule newSchedule = new Schedule(newDate,new Time(12,30,0),(int) centerId,(int) vaccineId,(int) lotSize);
```

Comment:
What happen if there is a restock of million shot restock?  Would you like a divide it into multiple schedule slots?